src/main/java/billgates/view/gui/SettingDialog.java

```diff
@@ -0,0 +1,124 @@
+            "Verdana", "Century", "Tahoma", "Copperplate"};
+    private static int colorIndex;
+    private static int fontIndex;
+    private boolean confirmed = false;
+    private final MainFrame mainFrame;
+
+    // Set up setting dialog to change background color and font style
+    public SettingDialog(MainFrame owner) {
+        super(owner);
+        mainFrame = owner;
+
+        // Set the font
+        this.setFont(new Font("Modern No. 20", Font.BOLD, 14));
+
+        // Set the Main root panel and its layout
+        JPanel root = new JPanel();
+        this.setTitle("Setting");
+        this.setModal(true);
+        this.setSize(400, 200);
+        this.setResizable(false);
+        SwingUtil.centerInOwner(this, this.getOwner());
+        this.setContentPane(root);
+
+        //Set the sub-action root panel
+        JPanel subAction = new JPanel();
+        subAction.add(confirmButton);
+        subAction.add(cancelButton);
+
+        // Set the color-field and font-field
+        JLabel backgroundButton = new JLabel("Background Color");
+        colorField.addItem("Default");
+        colorField.addItem("Red");
+        colorField.addItem("Orange");
+        colorField.addItem("Yellow");
+        colorField.addItem("Green");
+        colorField.addItem("Cyan");
+        colorField.addItem("Blue");
+        colorField.addItem("Pink");
+        colorField.addItem("White");
+
+        JLabel fontStyle = new JLabel("Font");
+        fontField.addItem("Default");
+        fontField.addItem("Times New Roman");
+        fontField.addItem("Helvetica");
+        fontField.addItem("Arial");
+        fontField.addItem("Impact");
+        fontField.addItem("Verdana");
+        fontField.addItem("Century");
+        fontField.addItem("Tahoma");
+        fontField.addItem("Copperplate");
+
+        //Set the color choose box panel
+        JPanel colorChooseBox = new JPanel();
+        colorChooseBox.add(backgroundButton);
+        colorChooseBox.add(colorField);
+
+        //Set the font choose box panel
+        JPanel fontChooseBox = new JPanel();
+        fontChooseBox.add(fontStyle);
+        fontChooseBox.add(fontField);
+
+        //Add the panels to root panel to set up our layout
+        root.add(colorChooseBox, BorderLayout.CENTER);
+        root.add(fontChooseBox, BorderLayout.CENTER);
+        root.add(subAction, BorderLayout.SOUTH);
+
+        // Case 1: Click cancelButton
+        cancelButton.addActionListener(e -> {
+            setVisible(false);
+        });
+
+        // Case 2: Choose different color and font
+        confirmButton.addActionListener((e ->
+        {
+            colorIndex = colorField.getSelectedIndex();
+            colorField.setSelectedIndex(colorIndex);
+
+            fontIndex = fontField.getSelectedIndex();
+            fontField.setSelectedIndex(fontIndex);
+
+            confirmed = true;
+
+            JOptionPane.showMessageDialog(null, "Page has been set!", "All Set", JOptionPane.PLAIN_MESSAGE);
```

Comment:
You can simply change `null` here to `this` and then the dialog will show in the center of the setting dialog no matter where the setting dialog currently is.